### PR TITLE
Make specviz viewer data retrieval methods consistent

### PR DESCRIPTION
**Pull request by ibusko**

### CODE REVIEW

**Review by javerbukh — APPROVED** [review]
`specviz.app.get_viewer("spectrum-viewer").data()[0]` and `specviz.get_spectra()['specviz_data|mZvT3&&Nemor;Z=iyd8f']` both returned the same value, looks good to me!

**Review by PatrickOgle — APPROVED** [review]
Verified that the two methods give equal flux, spectral_axis.  Didn't know how to evaluate InverseVariance function in Uncertainty of Spec1D.

**Comment by rosteen** [comment]
My only quibble with this is that it doesn't get subsets from the spectrum viewer, only BaseData, whereas `app.get_data_from_viewer("spectrum-viewer")` also returns subsets by default. 

Other than that I think this is fine for now, but long term it would be nice to have less duplicate code between this and `app.get_data_from_viewer` - as it stands currently there's no way to call that method from here, and it would be a bigger refactor to have `app.get_data_from_viewer` call the viewer `data()` methods. It all gets very circular!

File: jdaviz/configs/cubeviz/plugins/tests/test_data_retrieval.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import pytest
+import numpy as np
+
+from astropy.utils.data import download_file
+
+from jdaviz.app import Application
+
+URL = 'https://stsci.box.com/shared/static/28a88k1qfipo4yxc4p4d40v4axtlal8y.fits'
```
Comment (eteq) [inline jdaviz/configs/cubeviz/plugins/tests/test_data_retrieval.py:8]: Can this be changed to the "pretty" version of the URL? Or alternatively, a comment explaining which file this is?  Right now it's hard to look at this file and understand it.

Relatedly, can we get a quick docstring that says what the scope of this test file is?  I think "retreival" might be interpreted by some readers as being about the download, when this is really about "retrieving from glue" (and a docstring is sufficient to address that!)

**Review by eteq — APPROVED** [review]
One very minor thing, but oteherwise looks good!

**Comment by dtn5ah** [comment]
@ibusko Please see @rosteen's ticket before we merge this

**Comment by eteq** [comment]
> My only quibble with this is that it doesn't get subsets from the spectrum viewer, only BaseData, whereas app.get_data_from_viewer("spectrum-viewer") also returns subsets by default.

Clarification question: is there a ``jdaviz_app.get_viewer('spectrum-viewer').subset()`` or something like that? If so then maybe it's ok as-is and the problem is really that the name `get_data_...` is the problem (since "data" is an ambiguous name).  If not, then I agree fully with @rosteen

**Comment by rosteen** [comment]
The app method has a `include_subsets` named parameter that is `True` by default. The subsets are included in the returned dictionary keyed on their labels ("Subset 1", etc) along with the base data (e.g. "data whatever [FLUX]").

**Comment by dtn5ah** [comment]
(Just linking this PR to the corresponding issue) This resolves #206

**Comment by ibusko** [comment]
I added a fix for the subset retrieval bug. In  the process, I went ahead and re-instated the `include_subsets` keyword parameter to the `get_data_from_viewer` method. At least that way the behavior is consistent with the doc string. 

I note that there is quite a bit of code duplication in that solution. But it's not really copied-and-pasted code, since the app method returns a list with the base data and subsets, and the helper method returns a dict keyed by the subset name. Refactoring this is not a job for this iteration, I think (as @rosteen has pointed out already).

**Comment by javerbukh** [comment]
I do see that all methods now return subsets. However, when I look at the `spectral_axis` of the subsets, it's the same as the originating spectrum. I do not have much experience with subsets so this may be correct behavior but I just wanted to point that out. If that is as intended, then this should be good to merge.

**Comment by dtn5ah** [comment]
@javerbukh @ibusko and I met this morning and discussed that @javerbukh's observation is intentional; the data has the same spectral_axis because the subset simply applies a nan mask on top of the data not selected in the subset.

**Review by dtn5ah — APPROVED** [review]
I now see subsets in both retrieval methods. I also checked that the data between the subsets is the identical. Now one thing @javerbukh @ibusko and I noted in our meeting this morning is that the return types are different (list vs. dict). We note that 'consistent' in the title refers to the returning of subsets here, and not the return type. We will need to revisit this along with a deeper dive into an API refactor to fully address this

image

**Review by javerbukh — APPROVED** [review]
Agree with the previously mentioned comments. That said, this looks good to merge!

**Comment by javerbukh** [comment]
Merging now.

**Pull request merged**